revenger/infrastructure/python_adapter_v2.py

```diff
@@ -0,0 +1,212 @@
+                        if base_class == 'ABC':
+                            sub_datastructure.set_abstract()
+                        else:
+                            sub_datastructure.add_base_class(base_class)
+
+            return sub_datastructure, fqdn_class_name
+
+
+        def visit_ImportFrom(self, node: ImportFrom) -> Any:
+            module_path = node.module
+
+            for module_name in node.names:
+                self.from_import[module_name.name] = module_path + '.' + module_name.name
+
+            self.logger.log_warn(f' {self.from_import}')
+            ast.NodeVisitor.generic_visit(self, node)
+
+        def generic_visit(self, node):
+            # print(node)
+            ast.NodeVisitor.generic_visit(self, node)
+
+        def visit_ClassDef(self, node: ClassDef) -> Any:
+
+            sub_datastructure, fqdn_class_name = self.analyse_class_def(node)
+
+            self.active_class_name_list.append(fqdn_class_name)
+            ast.NodeVisitor.generic_visit(self, node)
+
+            if len(self.active_class_name_list) > 0:
+                self.active_class_name_list.pop()
+
+        def visit_AnnAssign(self, node: AnnAssign) -> Any:
+            static_name: str = node.target.id
+            static_type: str = CommonInfrastructure.NOT_EXTRACTED
+            if isinstance(node.annotation, ast.Name):
+                static_type = self.get_type(self.datastructure.get_skip_types(), \
+                                            node.annotation.id, self.from_import, self.filemodule)
+                self.logger.log_debug(f' Analyzing ast.Name static type {static_type} from file {self.filename}')
+            elif isinstance(node.annotation, ast.Subscript):
+                if isinstance(node.annotation.value, ast.Name) and \
+                        isinstance(node.annotation.slice, ast.Name):
+                    member_sub_type = self.get_type(self.datastructure.get_skip_types(), \
+                                                    node.annotation.slice.id, self.from_import, self.filemodule)
+                    static_type = node.annotation.value.id + '[' + \
+                                  member_sub_type + ']'
+                    self.logger.log_debug(
+                        f' Analyzing ast.Subscript static type {static_type} from file {self.filename}')
+            self.logger.log_debug(
+                f'   Static type from file {self.filename} found static_name: {static_name}, static_type: {static_type}')
+            sub_datastructure: GenericSubDataStructure = self.datastructure.get_datastructures_from_class_name(
+                self.get_current_class_name())
+
+            sub_datastructure.add_static(static_name, static_type)
+
+            # Instance and method vars
+            method: Datastructure.Method = sub_datastructure.get_method(self.active_method_name_list[-1])
+            target = node.target
+            if isinstance(target, ast.Attribute) or isinstance(target, ast.Name):
```

Comment:
I was wondering whether moving the check of attribute to a specific leave of the tree would make sense. What do you think?